### PR TITLE
NXP-28146 - Load nuxeo.NUXEO_ENVIRONMENT file in addition to nuxeo.defaults

### PR DESCRIPTION
**Pull request by kevinleturc**
T&P: https://qa.nuxeo.org/jenkins/job/TestAndPush/job/ondemand-testandpush-kleturc/715/

### CODE REVIEW

File: modules/runtime/nuxeo-launcher-commons/src/main/java/org/nuxeo/launcher/config/ServerConfigurator.java [hunk 1]
```diff
@@ -321,7 +365,10 @@ public void checkPaths() throws ConfigurationException {
             log.warn("NXP-8014 Packages cache location changed. You can safely delete {} or move its content to {}",
                     () -> oldPackagesPath, this::getPackagesDir);
         }
-
+        File oldPath = new File(getRuntimeHome(), "data" + File.separator + "vcsh2repo");
+        String message = String.format("NXP-5370, NXP-5460. " + "Please rename 'vcsh2repo' directory from %s to %s",
+                oldPath, new File(generator.getDataDir(), "h2" + File.separator + "nuxeo"));
```
Comment (efge) [inline modules/runtime/nuxeo-launcher-commons/src/main/java/org/nuxeo/launcher/config/ServerConfigurator.java:371]: This is really obsolete (I can't even remember when it was used). I don't think it's worth keeping.

File: modules/runtime/nuxeo-launcher-commons/src/main/java/org/nuxeo/launcher/config/ConfigurationGenerator.java [hunk 2]
```diff
@@ -1274,15 +1224,20 @@ public File getDataDir() {
...
+        createDirectoryIfDoesntExist(getTmpDir());
+        createDirectoryIfDoesntExist(getPackagesDir());
         checkAddressesAndPorts();
         serverConfigurator.verifyInstallation();
         backingServicesConfigurator.verifyInstallation();
+    }
 
+    private void createDirectoryIfDoesntExist(File directory) {
+        if (!directory.isDirectory()) {
+            directory.mkdirs();
```
Comment (efge) [inline modules/runtime/nuxeo-launcher-commons/src/main/java/org/nuxeo/launcher/config/ConfigurationGenerator.java:1251]: I think you can just call `mkdirs()` on all the dirs wihtout the `isDirectory` check, it shouldn't fail if the directory already exists.

File: modules/runtime/nuxeo-launcher-commons/src/main/java/org/nuxeo/launcher/config/ServerConfigurator.java [hunk 3]
```diff
@@ -111,16 +111,14 @@ public ServerConfigurator(ConfigurationGenerator configurationGenerator) {
...
-        return "conf" + File.separator + "Catalina" + File.separator + "localhost" + File.separator + getContextName()
-                + ".xml";
+        return generator.getNuxeoHome()
+                        .toPath()
+                        .resolve("conf")
+                        .resolve("Catalina")
+                        .resolve("localhost")
+                        .resolve(getContextName() + ".xml")
+                        .toFile()
+                        .exists();
```
Comment (efge) [inline modules/runtime/nuxeo-launcher-commons/src/main/java/org/nuxeo/launcher/config/ServerConfigurator.java:121]: Maybe `Files.exists(generator.getNuxeoHome().toPath().resolve(Path.of("conf", "Catalina", "localhost", getContextName() + ".xml")))`

File: modules/runtime/nuxeo-launcher-commons/src/main/java/org/nuxeo/launcher/config/ServerConfigurator.java (hunk 1 above)
Comment (kevinleturc) [reply modules/runtime/nuxeo-launcher-commons/src/main/java/org/nuxeo/launcher/config/ServerConfigurator.java]: I removed it.

**Pull request merged**